cfme/tests/infrastructure/test_providers.py

```diff
@@ -492,3 +495,56 @@ def test_infra_discovery_screen(appliance):
                    "to_ip4": ips['to']})
         view.start.click()
         view.flash.assert_message(ips['msg'])
+
+
+@pytest.fixture
+def setup_provider_min_templates(request, appliance, provider, min_templates):
+    templates_yaml = len(provider.data.get('templates', {}))
+    if templates_yaml < min_templates:
+        pytest.skip(f'Number of templates defined in yaml for {provider} does not meet minimum '
+                    f'for test parameter {min_templates}, skipping and not setting up provider')
+    if len(provider.mgmt.list_templates()) < min_templates:
+        pytest.skip(f'Number of templates on {provider} does not meet minimum '
+                    f'for test parameter {min_templates}, skipping and not setting up provider')
+    # Function-scoped fixture to set up a provider
+    setup_or_skip(request, provider)
+
+
+def verify_checked_items_compared(checkedList, view):
+    # The first and last header items do not contain template names, so are not iterated upon.
+    for header_with_template in view.comparison_table.headers[1:-1]:
+        try:
+            # Split and slice are used to remove extra characters in the header item
+            checkedList.remove(header_with_template.split(' ')[0])
+        except ValueError:
+            pytest.fail(f"Entity {header_with_template.split(' ')[0]} is in compare view, but was not checked.")
+        except TypeError:
+            pytest.fail('No entities found in compare view.')
+    if len(checkedList) > 0:
+        pytest.fail(f'Some checked items did not appear in the compare view: {checkedList}.')
+    return True
+
+
+@pytest.mark.parametrize("min_templates", [2, 4])
+@pytest.mark.meta(blockers=[BZ(1784180, forced_streams=["5.10"])], automates=[1784180])
```

Comment:
I think we can use a more restrictive provider filter here, its currently collecting `ONE_PER_VERSION` against `InfraProvider`.

As the test is only that the UI is including the selected templates, and does not validate the comparison table values, I think its safe to run it against `ONE` `InfraProvider`.

Reply:
Updated as suggested.